engine/class_modules/paladin/sc_paladin.cpp

```diff
@@ -2451,16 +2617,36 @@ double paladin_t::resource_gain( resource_e resource_type, double amount, gain_t
 
 double paladin_t::resource_loss( resource_e resource_type, double amount, gain_t* source, action_t* action )
 {
+  double initial_hp = health_percentage();
   double result = player_t::resource_loss( resource_type, amount, source, action );
   if (
       resource_type == RESOURCE_HOLY_POWER &&
       result > 0 &&
-      legendary.from_dusk_till_dawn -> ok() &&
-      resources.current[ RESOURCE_HOLY_POWER ] == legendary.from_dusk_till_dawn -> effectN( 2 ).base_value()
+      legendary.of_dusk_and_dawn -> ok() &&
+      resources.current[ RESOURCE_HOLY_POWER ] == legendary.of_dusk_and_dawn -> effectN( 2 ).base_value()
     )
   {
     buffs.blessing_of_dusk -> trigger();
   }
+
+  if (
+      resource_type == RESOURCE_HEALTH &&
+      legendary.reign_of_endless_kings -> ok() &&
+      ! buffs.reign_of_ancient_kings -> up() &&
+      health_percentage()/100 < legendary.reign_of_endless_kings -> effectN( 2 ).percent() &&
+      // Won't trigger if you're below the threshhold when the debuff expires. You have to be healed back over it for it to be able to proc again.
+      initial_hp/100 >= legendary.reign_of_endless_kings -> effectN( 2 ).percent()
```

Comment:
You could compare health_percentage() and initial_hp to the data's base_value() rather than dividing both by 100 and using percent()